lib/proxy/peer/quic.go

```diff
@@ -0,0 +1,141 @@
+// Teleport
+// Copyright (C) 2024 Gravitational, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package peer
+
+import (
+	"time"
+
+	"github.com/quic-go/quic-go/quicvarint"
+)
+
+const (
+	quicMaxIdleTimeout  = 30 * time.Second
+	quicKeepAlivePeriod = 5 * time.Second
+
+	quicMaxReceiveWindow = quicvarint.Max
+
+	// quicNextProto is the ALPN indicator for the current version of the QUIC
+	// proxy peering protocol.
+	quicNextProto = "teleport-peer-v1a"
```

Comment:
```suggestion
	quicNextProto = "teleport-peer-v1alpha1"
```

Reply:
Sure - although we should really think about shrinking our nextprotos rather than lengthen them; too big of a clienthello can be problematic 😬